idpi/test/test_flexpart.py

```diff
@@ -0,0 +1,239 @@
+import os
+import shutil
+import subprocess
+
+import cfgrib
+import cfgrib.xarray_to_grib
+import eccodes
+import jinja2
+import numpy as np
+import xarray as xr
+import yaml
+from operators.flexpart import fflexpart
+
+
+def get_da(field, dss):
+    for ds in dss:
+        if field in ds:
+            return ds[field]
+
+
+def load_data(fields, field_mapping, datafile):
+    ds = {}
+
+    read_keys = ["pv", "NV"]
+    dss = cfgrib.open_datasets(
+        datafile,
+        backend_kwargs={"read_keys": read_keys},
+        encode_cf=("time", "geography", "vertical"),
+    )
+
+    for f in fields:
+        ds[f] = get_da(field_mapping[f]["ifs"]["name"], dss)
+    ds["U"] = ds["U"].sel(hybrid=slice(40, 60))
+    ds["V"] = ds["V"].sel(hybrid=slice(40, 60))
+    ds["ETADOT"] = ds["ETADOT"].sel(hybrid=slice(1, 60))
+    ds["T"] = ds["T"].sel(hybrid=slice(40, 60))
+    ds["QV"] = ds["QV"].sel(hybrid=slice(40, 60))
+
+    return ds
+
+
+def test_flexpart():
+    gpaths = os.environ["GRIB_DEFINITION_PATH"].split(":")
+    eccodes_gpath = [p for p in gpaths if "cosmoDefinitions" not in p][0]
+    eccodes.codes_set_definitions_path(eccodes_gpath)
+
+    with open("field_mappings.yml") as f:
+        field_map = yaml.safe_load(f)
+
+    datadir = "/project/s83c/rz+/icon_data_processing_incubator/data/flexpart/"
+    datafile = datadir + "/efsf00000000"
+    constants = ("FIS", "FR_LAND", "SDOR")
+    inputf = (
+        "ETADOT",
+        "T",
+        "QV",
+        "U",
+        "V",
+        "PS",
+        "U_10M",
+        "V_10M",
+        "T_2M",
+        "TD_2M",
+        "CLCT",
+        "W_SNOW",
+        "TOT_CON",
+        "TOT_GSP",
+        "SSR",
+        "SSHF",
+        "EWSS",
+        "NSSS",
+    )
+
+    ds = load_data(constants + inputf, field_map, datafile)
+
+    for h in range(3, 10, 3):
+        datafile = datadir + f"/efsf00{h:02d}0000"
+        newds = load_data(inputf, field_map, datafile)
+
+        for field in newds:
+            ds[field] = xr.concat([ds[field], newds[field]], dim="step")
+
+    NV = ds["U"].GRIB_NV
```

Comment:
Add some comments on the steps that are implemented here.

Reply:
I moved all this to a function. But reminder, I would like to discuss this with you before merge